librosa/core/pitch.py

```diff
@@ -959,15 +946,28 @@ def __check_yin_params(
     if fmin <= 0:
         raise ParameterError(f"fmin={fmin:.3f} must be strictly positive")
 
-    if win_length >= frame_length:
+    if sr / fmin >= frame_length - 1:
```

Comment:
does this PR introduce `ParameterError`'s for cases which were did not error (at least for some values of `win_length`) up until today?

Reply:
In the current state of this PR, `win_length` is actually completely ignored. A deprecation warning (not an error) will be logged if `yin` or `pyin` is called with `win_length` set to any value other than `None`.

I do think the `win_length` parameter should be discarded entirely going forward, and I think this way it's not breaking anyone's existing code.

However, I think it might introduce `ParameterError`s for some values of `fmin` and `fmax`. I believe the current librosa code erroneously checks `fmax` to determine whether the largest period fits into the frame, which this PR changes to checking `fmin` (lowest frequency ≘ largest period).

Reply:
If `win_length` is deprecated, please use the `util.deprecation.Deprecated` helper class to mark it as such.  This way we can distinguish between an explicit `None` setting (which would be an error going forward) and using the default (which would not be an error in the future).